lib/jekyll/renderer.rb

```diff
@@ -13,78 +13,96 @@ def initialize(site, document, site_payload = nil)
     # Determine which converters to use based on this document's
     # extension.
     #
-    # Returns an array of Converter instances.
+    # Returns Array of Converter instances.
     def converters
-      @converters ||= site.converters.select { |c| c.matches(document.extname) }
+      @converters ||= site.converters.select { |c| c.matches(document.extname) }.sort
     end
 
     # Determine the extname the outputted file should have
     #
-    # Returns the output extname including the leading period.
+    # Returns String the output extname including the leading period.
     def output_ext
       @output_ext ||= (permalink_ext || converter_output_ext)
     end
 
-    ######################
-    ## DAT RENDER THO
-    ######################
-
+    # Prepare payload and render the document
+    #
+    # Returns String rendered document output
     def run
       Jekyll.logger.debug "Rendering:", document.relative_path
 
-      payload["page"] = document.to_liquid
+      assign_pages
+      assign_related_posts
+      assign_highlighter_options
+
+      Jekyll.logger.debug "Pre-Render Hooks:", document.relative_path
+      document.trigger_hooks(:pre_render, payload)
+
+      render_document
+
+      document.output
+    end
 
-      if document.respond_to? :pager
+    # Set page content to payload and assign pager if document has one.
+    #
+    # Returns nothing
+    private
+    def assign_pages
+      payload["page"] = document.to_liquid
+      if document.respond_to?(:pager)
         payload["paginator"] = document.pager.to_liquid
       end
+    end
 
+    # Set related posts to payload if document is a post.
+    #
+    # Returns nothing
+    private
+    def assign_related_posts
       if document.is_a?(Document) && document.collection.label == "posts"
         payload["site"]["related_posts"] = document.related_posts
       else
         payload["site"]["related_posts"] = nil
       end
+    end
 
-      # render and transform content (this becomes the final content of the object)
+    # Set highlighter prefix and suffix
+    #
+    # Returns nothing
+    private
+    def assign_highlighter_options
       payload["highlighter_prefix"] = converters.first.highlighter_prefix
       payload["highlighter_suffix"] = converters.first.highlighter_suffix
+    end
 
-      Jekyll.logger.debug "Pre-Render Hooks:", document.relative_path
-      document.trigger_hooks(:pre_render, payload)
-
-      info = {
-        :registers => { :site => site, :page => payload["page"] }
-      }
-
-      output = document.content
-
+    # Render the document.
+    # Puts rendered content into document.output
+    #
+    # Returns nothing
+    private
+    # rubocop: disable AbcSize
```

Comment:
The ABC metric is 25 in this method (20 is max).
I couldn't think of a way to reduce it without creating too many helper methods.
Any suggestions? 😄